src/Soneta.Sdk/Sdk/Sdk.targets

```diff
@@ -18,6 +26,152 @@
+
+        public override bool Execute()
+        {
+            CurrentVersion = GetCurrentVersion();
+            LatestVersion = GetLatestVersion();
+            IsUpdateAvailable = CheckIfUpdateAvailable(CurrentVersion, LatestVersion);
+
+            return true;
+        }
+
+        private string GetCurrentVersion()
+        {
+            if (File.Exists(GlobalJsonFilePath))
+            {
+                var json = File.ReadAllText(GlobalJsonFilePath);
+                return GetMatchingValue(json, GlobalJsonSdkVersionPattern);
+            }
+
+            return null;
+        }
+
+        private string GetMatchingValue(string json, string pattern)
+        {
+            var match = Regex.Match(json, pattern);
+            return match.Value;
+        }
+
+        private string GetLatestVersion()
+        {
+            return CalculateUpdateCheckTime() < DateTime.Now ? 
+                GetRemoteVersion() : GetLocalVersion();
+        }
+
+        private DateTime CalculateUpdateCheckTime() 
+        {
+            var lastWriteTime = File.GetLastWriteTime(LatestVersionFilePath);
+            Enum.TryParse(UpdateIntervalUnit, out TimeUnit unit);
+
+            switch (unit)
+            {
+                case TimeUnit.Seconds:
+                    return lastWriteTime.AddSeconds(UpdateInterval);
+                case TimeUnit.Minutes:
+                    return lastWriteTime.AddMinutes(UpdateInterval);
+                case TimeUnit.Hours:
+                    return lastWriteTime.AddHours(UpdateInterval);
+                case TimeUnit.Days:
+                    return lastWriteTime.AddDays(UpdateInterval);
+                case TimeUnit.Weeks:
+                    return lastWriteTime.AddDays(UpdateInterval * 7);
+                case TimeUnit.Months:
+                    return lastWriteTime.AddMonths(UpdateInterval);
+                case TimeUnit.Years:
+                    return lastWriteTime.AddYears(UpdateInterval);
+                default:
+                    return DateTime.MaxValue;
+            }
+        }
+
+        private string GetRemoteVersion()
+        {
+            var latestVersion = GetLatestVersionFromApi();
+            StoreLatestVersion(latestVersion);
+            return latestVersion;
+        }
+
+        private string GetLatestVersionFromApi()
+        {
+            using (var httpClient = new HttpClient())
+            {
+                var json = httpClient.GetStringAsync(NugetApiUrl).Result;
+                return GetMatchingValue(json, NugetApiLatestVersionPattern);
+            }
+        }
+
+        private void StoreLatestVersion(string version) 
+        {
+            using (var sw = File.CreateText(LatestVersionFilePath))
+                sw.WriteLine(version);
+        }
+
+        private string GetLocalVersion() 
+        {
+            return File.ReadAllText(LatestVersionFilePath).Trim();
```

Comment:
Podobnie jak wyżej, czy w przypadku gdy plik nie istnieje, czy przypadkiem nie poleci coś w stylu: "Could not find a part of the path..."

Reply:
Jeśli kod dojdzie do tego miejsca, to plik ten (prawie) na pewno będzie istniał, ponieważ wywoła się logika o której piszę w komentarzu powyżej. Podobnie jak w tamtym przypadku pozostawiam do dyskusji, czy dodać tu jawne sprawdzenie istnienia pliku.